### DISCUSSION TITLE
Readable formatting of URL Query Params (JSON→URL)

### DISCUSSION THREAD

**Discussion by AlonMiz** [TanStack/router#932] [Ideas]
the ability to have strong control of the query param has always been a crucial point in the apps I build. 
It's important to manage the state and not duplicate it with the internal state. following the KISS principle.

I would point out that, in addition to being pure state management, it should also be readable.

thus I think that the default SearchParam formatting can be improved, there are better ways to do it. like json<->url
https://jsonurl.org/
imagine:
`{"Hello":"World!", "some": {"nested": [1,2,3]}}` => `(Hello:World!,some:(nested:(1,2,3)))`
![image](https://github.com/TanStack/router/assets/1812912/8709b5e3-daeb-4849-9d30-f14f9f4b0c40)

much cleaner then `'%7B%22Hello%22:%22World!%22,%20%22some%22:%20%7B%22nested%22:%20%5B1,2,3%5D%7D%7D'`

What do you think?

**Comment by schiller-manuel** [comment]
You probably already saw this, right? https://tanstack.com/router/v1/docs/guide/custom-search-param-serialization

**Reply by AlonMiz** [reply]
i didn't 🙈 
so considering this,integrating json->URL is super easy!
👏
